library/std/src/sys/pal/unix/stack_overflow.rs

```diff
@@ -86,13 +86,21 @@ mod imp {
     // out many large systems and all implementations allow returning from a
     // signal handler to work. For a more detailed explanation see the
     // comments on #26458.
+    /// SIGSEGV/SIGBUS entry point
+    /// # Safety
+    /// Rust doesn't call this, it *gets called*.
+    #[forbid(unsafe_op_in_unsafe_fn)]
     unsafe extern "C" fn signal_handler(
         signum: libc::c_int,
         info: *mut libc::siginfo_t,
         _data: *mut libc::c_void,
     ) {
         let (start, end) = GUARD.get();
-        let addr = (*info).si_addr() as usize;
+        if info.is_null() || !info.is_aligned() {
+            rtabort!("signal handler passed null/non-aligned siginfo_t ptr")
+        };
+        // SAFETY: we've done our due diligence, so assume the pointer is to a real siginfo_t
```

Comment:
The pointer is provided by the system and will always fulfil these requirements. We don't need to check for bugs in the kernel in `std`, if these checks fail, the system is fundamentally broken.
```suggestion
       // SAFETY: this pointer is provided by the system and will always point to a valid `siginfo_t`.
```

Reply:
...Alright but I'm just thinking rn about how many times I've seen that statement pan out to "so, in this one case,"